flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/enumerator/MySqlSourceEnumerator.java

```diff
@@ -121,6 +122,17 @@ public void handleSplitRequest(int subtaskId, @Nullable String requesterHostname
     public void addSplitsBack(List<MySqlSplit> splits, int subtaskId) {
         LOG.debug("MySQL Source Enumerator adds splits back: {}", splits);
         splitAssigner.addSplits(splits);
+
+        if (isNewlyAddedAssigningFinished(splitAssigner.getAssignerStatus())) {
```

Comment:
Thanks @lzshlzsh for the contribution, but I recently improve the newly added table process. Could you help check is the issue still exits in current codebase?

Reply:
I see #1927，it is a great job. I will take a closer look at it and verify it.  Thanks !

Reply:
@leonardBang After carefully reviewing and testing the non blocking newly add table process #1927, I found a problem, perhaps a design trade-off. Could you explain it in detail？
The in turn execution of BinlogSplitReader and SnapshotSplitReader has achieved the goal of non blocking, but the data read by BinlogSplitReader has not been modified to filter out the data of the newly added table. This may not be a problem in the ETL scenario, but it can lead to incorrect results in scenarios such as group by.

Test case：
1、do savepoint
2、Insert (2001, 17, da) into the newly added table, table schema is:
```
CREATE TABLE `mysql_cdc`
(
    `f_sequence`   INT,
    `f_random`     INT,
    `f_random_str` VARCHAR,
    PRIMARY KEY (`f_sequence`) NOT ENFORCED
) WITH (
  'connector' = 'mysql-cdc',
  ...
);
```
3、restore from savepoint，output duplicated data of key 2001. One emitted by BinlogSplitReader, and the other by SnapshotSplitReader.
```
 +I[2001, 17, da]
 +I[2001, 17, da]
```

Reply:
I didn't reproduce this case @lzshlzsh , and the case should be covered by NewlyAddedTableITCase#testNewlyAddedTableForExistsPipelineSingleParallelismWithAheadBinlog()